diff --git a/src/BulkUpdateFields.php b/src/BulkUpdateFields.php
index 1234567..abcdefg 100644
--- a/src/BulkUpdateFields.php
+++ b/src/BulkUpdateFields.php
@@ -2,6 +2,7 @@

 namespace Drupal\bulk_update_fields;

+use Drupal\Core\Extension\ModuleHandlerInterface;
 use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
 use Drupal\paragraphs\Entity\Paragraph;

@@ -37,6 +38,10 @@ class BulkUpdateFields {
       }
     }

+    /** @var ModuleHandlerInterface $moduleHandler */
+    $moduleHandler = \Drupal::service('module_handler');
+    $moduleHandler->alter('bulk_update_fields_process_field', $value, $field_definition);
+
     return $value;
   }

@@ -57,6 +62,10 @@ class BulkUpdateFields {
       $field_value = $field_value['value'];
     }

+    /** @var ModuleHandlerInterface $moduleHandler */
+    $moduleHandler = \Drupal::service('module_handler');
+    $moduleHandler->alter('bulk_update_fields_preprocess_field', $field_value);
+
     return $field_value;
   }

diff --git a/src/bulk_update_fields.api.php b/src/bulk_update_fields.api.php
new file mode 100644
index 0000000..fbb643c
--- /dev/null
+++ b/src/bulk_update_fields.api.php
@@ -0,0 +1,24 @@
+<?php
+
+/**
+ * Alter a field value before processessing and saving to an entity
+ *
+ * @param $field_Value
+ *  The array representation of the field value before saving
+ */
+function hook_bulk_update_fields_preprocess_field_alter(&$field_value) {
+  if (isset($field_value['add_more'])) {
+    unset($field_value['add_more']);
+  }
+}
+
+/**
+ * Alter a single field value before being saved to an entity
+ *
+ * @param $value
+ *  A single field value to save
+ * @param $field_definition
+ *  The definition of the field to be saved
+ */
+function hook_bulk_update_fields_process_field_alter(&$value, $field_definition) {
+}
