diff --git a/core/modules/link/src/Plugin/Field/FieldWidget/LinkWidget.php b/core/modules/link/src/Plugin/Field/FieldWidget/LinkWidget.php
index 37acb8c16..46a3ed1d5 100644
--- a/core/modules/link/src/Plugin/Field/FieldWidget/LinkWidget.php
+++ b/core/modules/link/src/Plugin/Field/FieldWidget/LinkWidget.php
@@ -123,7 +123,7 @@ protected static function getUserEnteredStringAsUri($string) {
       $uri = 'route:' . $string;
     }
     // Detects if this string is a replaceable token.
-    elseif (\Drupal::token()->scan($string)) {
+    elseif (strpos($string, "[") === 0) {
       if (strpos($string, "token:") === FALSE) {
         $uri = 'token:' . $string;
       }
