diff --git a/show_title.module b/show_title.module
index ec36538..7998d6e 100644
--- a/show_title.module
+++ b/show_title.module
@@ -205,10 +205,23 @@ function show_title_theme_registry_alter(&$theme_registry) {
     }
 
     // Set the branding block template.
-    $module_path = \Drupal::service('extension.list.module')->getPath('show_title');
-    $theme_registry['node']['path'] = implode(
-      '/',
-      [$module_path, 'templates', $base_theme]
-    );
+    $theme_registry['node']['path'] = _show_title_get_base_theme_node_template($base_theme);
   }
 }
+
+function _show_title_get_base_theme_node_template($themeName)
+{
+  $module_path = \Drupal::service('extension.list.module')->getPath('show_title');
+  $templatePath = implode('/',
+    [$module_path, 'templates', $themeName]
+  );
+  if (!is_dir($templatePath)) {
+    $theme = \Drupal::service('theme_handler')->getTheme($themeName);
+    if (!empty($theme->base_theme)) {
+      return _show_title_get_base_theme_node_template($theme->base_theme);
+    } else {
+      return _show_title_get_base_theme_node_template('stable');
+    }
+  }
+  return $templatePath;
+}
